### PR TITLE
detect/flow: fix DETECT_FLOW_FLAG_ESTABLISHED check

### PR DESCRIPTION
**Pull request by open-cmi**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://suricata.readthedocs.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
- detect/flow: fix DETECT_FLOW_FLAG_ESTABLISHED check

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the pull request number.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Review by inashivb — CHANGES_REQUESTED** [review]
Hi! Thanks for your first contribution to our project. :)
1. Could you please fix your commit message to something like
`detect/flow: fix DETECT_FLOW_FLAG_ESTABLISHED check`
or whatever you feel makes sense but the prefix should indicate the part of code being changed.
2. Also, please make sure to sign the contribution agreement for your work to be accepted.

**Comment by open-cmi** [comment]
> Hi! Thanks for your first contribution to our project. :)
> 
> 1. Could you please fix your commit message to something like
>    `detect/flow: fix DETECT_FLOW_FLAG_ESTABLISHED check`
>    or whatever you feel makes sense but the prefix should indicate the part of code being changed.
> 2. Also, please make sure to sign the contribution agreement for your work to be accepted.
> 3. Your name in commit sign should be `First-name Last-name`

OK, I had modified the commit message

**Comment by victorjulien** [comment]
It would be great if you'd also create some Suricata-Verify tests to show the issue, so that we can continue to test for it in the future.

**Comment by open-cmi** [comment]
> It would be great if you'd also create some Suricata-Verify tests to show the issue, so that we can continue to test for it in the future.

OK, I add a unit test for test setting establish and not_establish both.

**Comment by victorjulien** [comment]
> > It would be great if you'd also create some Suricata-Verify tests to show the issue, so that we can continue to test for it in the future.
> 
> I added an unit test for test setting establish and not_establish both. This is my first time to write an unit test, is it what you want?

I think the UT is useful, and can stay as part of the PR, but I meant Suricata-Verify: https://github.com/OISF/suricata-verify
The benefit of SV is that they are tested against 6.0.x as well and additionally act as "seed" to the fuzzing effort.

**Comment by victorjulien** [comment]
Additionally, please make sure the formatting check passes.

**Comment by open-cmi** [comment]
> Additionally, please make sure the formatting check passes.

OK

**Comment by open-cmi** [comment]
> > > It would be great if you'd also create some Suricata-Verify tests to show the issue, so that we can continue to test for it in the future.
> > 
> > 
> > I added an unit test for test setting establish and not_establish both. This is my first time to write an unit test, is it what you want?
> 
> I think the UT is useful, and can stay as part of the PR, but I meant Suricata-Verify: https://github.com/OISF/suricata-verify The benefit of SV is that they are tested against 6.0.x as well and additionally act as "seed" to the fuzzing effort.

OK, I will learn how to write a SV test. Thanks.

File: src/detect-flow.c [hunk 1]
```diff
@@ -943,6 +943,17 @@ static int DetectFlowTestParse21 (void)
     PASS;
 }
 
+/**
+ * \test DetectFlowTestParse22 is a test for setting the established,not_established flow opts both
+ */
+static int DetectFlowTestParse22(void)
+{
+    DetectFlowData *fd = NULL;
+    fd = DetectFlowParse(NULL, "established,not_established");
```
Comment (catenacyber) [inline src/detect-flow.c:952]: Also `not_established ,established` should have the same error message

**Pull request closed without merging**